src/helix/core.clj

```diff
@@ -44,19 +50,30 @@
                     (:native (meta type)))
         type (if (keyword? type)
                (name type)
-               type)]
-    (cond
-      (map? (first args))
-      `^js/React.Element (.createElement
-                          (get-react)
-                          ~type
-                          ~(if native?
-                             `(impl.props/dom-props ~(first args))
-                             `(impl.props/props ~(first args)))
-                          ~@(rest args))
-
-      :else `^js/React.Element (.createElement (get-react) ~type nil ~@args))))
-
+               type)
+        has-props? (or (map? (first args))
+                       (nil? (first args)))
+        children (if has-props?
+                   (rest args)
+                   args)
+        props (if (map? (first args))
+                (if native?
+                  `(impl.props/dom-props ~(first args) ~(jsx-children children))
+                  `(impl.props/props     ~(first args) ~(jsx-children children)))
+                (tl/->JSValue (cond-> {}
+                                (not-empty children)
+                                (assoc :children (jsx-children children)))))
+        has-key? (when has-props?
+                   (contains? (first args) :key))
+        the-key (when has-key?
+                  (:key (first args)))
+        emit-fn (if (and (seq? children)
+                         (next children))
```

Comment:
I think this could be simplified

```suggestion
        emit-fn (if (next children)
```

Reply:
huh, reading the code I thought that at this point `children` is a seq, either `args` (seq from rest params) or `(rest args)`, in which case I think it would be safe to just call `next`.